egs/wsj/s5/steps/dur_model/get_examples.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+
+# Copyright 2015 Hossein Hadian
+
+cmd=run.pl
+validation_ratio=5
+shuffle_buffer_size=5000
+
+[ -f ./path.sh ] && . ./path.sh; # source the path.
+. parse_options.sh || exit 1;
+
+echo "$0 $@"  # Print the command line for logging
+
+if [ $# != 2 ]; then
+   echo "Usage: $0 [options] <ali-dir> <duration-model-dir>"
+   echo "e.g.: $0 exp/mono_ali exp/mono/durmod"
+   echo ""
+   echo "Main options (for others, see top of script file):"
+   echo "  --validation-ratio      <number>                 # ratio of validation data over all data. default is 5 (i.e. 5 percent)"
+   exit 1;
+fi
+
+alidir=$1
+dir=$2
+
+durmodel=$dir/durmodel.mdl
+transmodel=$alidir/final.mdl
+
+for f in $transmodel $alidir/ali.1.gz; do
+  [ ! -f $f ] && echo "$0: Required file not found: $f" && exit 1;
+done
+
+if [ ! -f $dir/all.ali ]; then
+  gunzip -c $alidir/ali.*.gz >> $dir/all.ali || exit 1;
+fi
+
+$cmd $dir/log/durmod_make_examples.log \
+     nnet3-durmodel-make-egs $durmodel $transmodel ark:$dir/all.ali ark,t:$dir/all.egs || exit 1;
```

Comment:
You could directly support the use of 'ark:gunzip -c $alidir/ali.*.gz |' rather than creating a temporary all.ali file